src/compiler/app/app-core-esm.ts

```diff
@@ -6,35 +6,33 @@ import { generateAppGlobalScript } from './app-global-scripts';
 import { setBuildConditionals } from './build-conditionals';
 
 
-export async function generateEsmCore(config: d.Config, compilerCtx: d.CompilerCtx, buildCtx: d.BuildCtx, outputTarget: d.OutputTarget, entryModules: d.EntryModule[], appRegistry: d.AppRegistry) {
-  if (outputTarget.type !== 'dist') {
-    return;
-  }
-
-  // mega-minify the core w/ property renaming, but not the user's globals
-  // hardcode which features should and should not go in the core builds
-  // process the transpiled code by removing unused code and minify when configured to do so
-  let jsContent = await config.sys.getClientCoreFile({ staticName: 'core.esm.js' });
+export async function generateEsmCore(config: d.Config, compilerCtx: d.CompilerCtx, buildCtx: d.BuildCtx, outputTarget: d.OutputTargetBuild, entryModules: d.EntryModule[], appRegistry: d.AppRegistry) {
+  if (outputTarget.type === 'dist') {
```

Comment:
this way, TS can infer the real type of OutputTarget, by using discriminated types: https://basarat.gitbooks.io/typescript/docs/types/discriminated-unions.html